tests/unit/functions/edh-marshaller.test.ts

```diff
@@ -1,14 +1,39 @@
 import { edhMarshaller } from '../../../src/functions/edh-marshaller';
-import { SQSService } from '../../../src/services/sqs';
 import { Context } from 'aws-lambda';
 import * as _ from '../../../src/utils/transform-tech-record';
+import AWSXray from 'aws-xray-sdk';
+import { SQS, AWSError, Service, Response } from 'aws-sdk';
+import { SendMessageRequest } from 'aws-sdk/clients/sqs';
+import { BatchItemFailuresResponse } from '../../../src/models/BatchItemFailures';
+
+type PromiseResult<D, E> = D & { $response: Response<D, E> };
+
+const mSendMessage = jest.fn();
+
+class SqsMock extends Service {
+  sendMessage(params: SendMessageRequest) {
+    mSendMessage(params);
+
+    return { 
+      promise: jest.fn().mockImplementation(() => {
+        if (params.QueueUrl === 'FAIL') {
+          return Promise.reject(<PromiseResult<SQS.SendMessageResult, AWSError>>{});
+        }
+        return Promise.resolve(<PromiseResult<SQS.SendMessageResult, AWSError>>{});
+      }),
+    };
```

Comment:
Just to make it a bit cleaner:
```suggestion
      promise: jest.fn().mockImplementation(() => params.QueueUrl === 'FAIL'
        ? Promise.reject(<PromiseResult<SQS.SendMessageResult, AWSError>>{})
        : Promise.resolve(<PromiseResult<SQS.SendMessageResult, AWSError>>{})),
    };
```